From cdd4c2ce2a8b54ba70f4422070fa3d7e7df77119 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Storsj=C3=B6?= <martin@martin.st>
Date: Wed, 2 Nov 2022 11:55:39 +0000
Subject: [PATCH 03/11] [openmp] [test] Fix building the
 affinity/format/fields_values.c testcase on Windows

Add a missing <process.h> include for _getpid. Don't typedef the
pid_t type on mingw, as mingw headers already provide a typedef for
it.

Differential Revision: https://reviews.llvm.org/D137745
---
 openmp/runtime/test/affinity/format/fields_values.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/openmp/runtime/test/affinity/format/fields_values.c b/openmp/runtime/test/affinity/format/fields_values.c
index e56ce2786be4..e7c1faf3e3c6 100644
--- a/openmp/runtime/test/affinity/format/fields_values.c
+++ b/openmp/runtime/test/affinity/format/fields_values.c
@@ -19,8 +19,11 @@
 
 #if defined(_WIN32)
 #include <windows.h>
+#include <process.h>
 #define getpid _getpid
+#ifndef __MINGW32__
 typedef int pid_t;
+#endif
 #define gettid GetCurrentThreadId
 #define my_gethostname(buf, sz) GetComputerNameA(buf, &(sz))
 #else
-- 
2.25.1

